### ISSUE TITLE
The command '/bin/sh -c ln -s /taiga/conf.json /usr/src/taiga-front-dist/dist/conf.json' returned a non-zero code: 1

### ISSUE THREAD

**Issue by gerroon** [benhutchins/docker-taiga#72]
Hi

Trying to set this up on Debian Buster but I keep hitting errors, one of it is below. 

```
Generation complete.
Setting up gettext (0.19.8.1-2) ...
+ rm -rf /var/lib/apt/lists/deb.debian.org_debian_dists_stretch-updates_InRelease /var/lib/apt/lists/deb.debian.org_debian_dists_stretch-updates_main_binary-amd64_Packages.lz4 /var/lib/apt/lists/deb.debian.org_debian_dists_stretch_Release /var/lib/apt/lists/deb.debian.org_debian_dists_stretch_Release.gpg /var/lib/apt/lists/deb.debian.org_debian_dists_stretch_main_binary-amd64_Packages.lz4 /var/lib/apt/lists/lock /var/lib/apt/lists/nginx.org_packages_mainline_debian_dists_stretch_InRelease /var/lib/apt/lists/nginx.org_packages_mainline_debian_dists_stretch_nginx_binary-amd64_Packages.lz4 /var/lib/apt/lists/partial /var/lib/apt/lists/security.debian.org_debian-security_dists_stretch_updates_InRelease /var/lib/apt/lists/security.debian.org_debian-security_dists_stretch_updates_main_binary-amd64_Packages.lz4
Removing intermediate container 6587aee3f929
 ---> 42ec0a19ddcf
Step 8/47 : RUN locale-gen en_US.UTF-8 && dpkg-reconfigure locales
 ---> Running in 583ff46d327a
Generating locales (this might take a while)...
Generation complete.
Generating locales (this might take a while)...
Generation complete.
Removing intermediate container 583ff46d327a
 ---> df4aebb1faf4
Step 9/47 : COPY taiga-back /usr/src/taiga-back
 ---> 0887c1ebdfca
Step 10/47 : COPY taiga-front-dist/ /usr/src/taiga-front-dist
 ---> b4d4165d16da
Step 11/47 : COPY docker-settings.py /usr/src/taiga-back/settings/docker.py
 ---> c1dbd7a9328e
Step 12/47 : COPY conf/locale.gen /etc/locale.gen
 ---> 2b84a07cc5ed
Step 13/47 : COPY conf/nginx/nginx.conf /etc/nginx/nginx.conf
 ---> f7aeefa52ef7
Step 14/47 : COPY conf/nginx/taiga.conf /etc/nginx/conf.d/default.conf
 ---> 4d9d8a5839e0
Step 15/47 : COPY conf/nginx/ssl.conf /etc/nginx/ssl.conf
 ---> 1156c312d5f4
Step 16/47 : COPY conf/nginx/taiga-events.conf /etc/nginx/taiga-events.conf
 ---> 1d9b55ae51bd
Step 17/47 : RUN mkdir -p /taiga
 ---> Running in aaa5afd57262
Removing intermediate container aaa5afd57262
 ---> df4b349624d5
Step 18/47 : COPY conf/taiga/local.py /taiga/local.py
 ---> b022e9827c9d
Step 19/47 : COPY conf/taiga/conf.json /taiga/conf.json
 ---> b4b55ac4f77f
Step 20/47 : RUN ln -s /taiga/local.py /usr/src/taiga-back/settings/local.py
 ---> Running in 1358fee11d19
Removing intermediate container 1358fee11d19
 ---> 5e7bdd4f3e14
Step 21/47 : RUN ln -s /taiga/conf.json /usr/src/taiga-front-dist/dist/conf.json
 ---> Running in d808da500ecb
ln: failed to create symbolic link '/usr/src/taiga-front-dist/dist/conf.json': No such file or directory
ERROR: Service 'taiga' failed to build: The command '/bin/sh -c ln -s /taiga/conf.json /usr/src/taiga-front-dist/dist/conf.json' returned a non-zero code: 1
```

**Comment by BiBzz** [comment]
Hi @gerroon,

I faced the same. It have been answered by @benhutchins in #16 : 
> It looks like you are building, but didn't pull down the submodules.
> Try running:
> `git submodule update --init`

**Comment by gerroon** [comment]
Thanks but not sure, I still have issues with the compose :(

**}'
     void X509_get0_signature(const ASN1_BIT_STRING **psig,
          ^~~~~~~~~~~~~~~~~~~
    build/temp.linux-x86_64-3.5/_openssl.c:62516:29: warning: passing argument 2 of 'X509_get0_signature' from incompatible pointer type [-Wincompatible-pointer-types]
       { X509_get0_signature(x0, x1, x2); }
                                 ^~
    In file included from /usr/include/openssl/cms.h:16:0,
                     from build/temp.linux-x86_64-3.5/_openssl.c:545:
    /usr/include/openssl/x509.h:552:6: note: expected 'const X509_ALGOR ** {aka const struct X509_algor_st **}' but argument is of type 'X509_ALGOR ** {aka struct X509_algor_st **}'
     void X509_get0_signature(const ASN1_BIT_STRING **psig,
          ^~~~~~~~~~~~~~~~~~~
    At top level:
    build/temp.linux-x86_64-3.5/_openssl.c:3426:13: warning: '_ssl_thread_locking_function' defined but not used [-Wunused-function]
     static void _ssl_thread_locking_function(int mode, int n, const char *file,
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-vs2kqvkg/cryptography/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-oiw67hj1/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-install-vs2kqvkg/cryptography/

**Comment by BiBzz** [comment]
Yeah sorry, faced this one too. We're following the exactly same path ^^

This error comes from Taiga Back and seems to be versions incompatibility between openssl and cryptography (source: https://github.com/taigaio/taiga-back/issues/1003). Thanks to new requirements this issue had been fixed in Taiga Back so my quick fix was to update submodules with latest versions with #70.
> Ideally, the process would only require you to run:
> 
> `git submodule update --init --remote`
> 
> However, based on some other issues I think updating Nginx and some other aspects of this project may be required. I'm going through issues and pull requests now, but you will likely need to pull the newer image.
That was the solution for me to overcome this issue.

Now I get no more error (hoorray), but I face the taiga "an issue occurred" page (shame). Perhaps a configuration forgot. I'll look at it this later today.
Hope you'll can bring a fresh look on this.

**Comment by janezgec** [comment]
I encountered exact same issues when setting up taiga, thank you @BiBzz ... 

For the issue mentioned at the end with the "an issue occured" page.
If you are using https, make sure your docker-compose configuration, contains:
```
TAIGA_SSL: 'True' # To enable SSL, uncomment this line
      
TAIGA_SSL_BY_REVERSE_PROXY: 'True' # To enable SSL, handling by a reverse proxy, uncomment this
```

Also, make sure the values are "True" not lowercase "true" - that doesnt work.

**Comment by BiBzz** [comment]
Thanks @zgec for having pointing this!
I deleted `TAIGA_SSL_BY_REVERSE_PROXY` by mistake when I merge docker-compose.yml
